traefik/tests/service-config_test.yaml

```
@@ -222,3 +222,17 @@ tests:
       - equal:
           path: items[1].spec.ipFamilyPolicy
           value: PreferDualStack
+  - it: should set the service port to advertisedPort for HTTP/3 services
+    set:
+      image:
+        tag: 2.6.0
+      ports:
+        websecure:
+          http3:
+            advertisedPort: 8843
+          tls:
+            enabled: true
+    asserts:
+    - equal:
+        path: items[1].spec.ports[0].port
+        value: 8843
```

Maybe we should also add a check for it to be UDP
```suggestion
    - equal:
        path: items[1].spec.ports[0].port
        value: 8843
    - equal:
        path: items[1].spec.ports[0].protocol
        value: UDP
```